core/src/main/java/org/apache/brooklyn/util/core/xstream/XmlSerializer.java

```diff
@@ -80,6 +80,11 @@ protected MapperWrapper wrapMapper(MapperWrapper next) {
 
         xstream.registerConverter(new EnumCaseForgivingConverter());
         xstream.registerConverter(new Inet4AddressConverter());
+        
+        // See ObjectWithDefaultStringImplConverter (and its usage) for why we want to auto-detect 
+        // annotations (usages of this is in the camp project, so we can't just list it statically
+        // here unfortunately).
+        xstream.autodetectAnnotations(true);
```

Comment:
From http://x-stream.github.io/annotations-tutorial.html#AutoDetect

> In auto-detection mode XStream will have to examine any unknown class type for annotations. This will slow down the marshalling process until all processed types have been examined once.